src/Uno.UI/UI/Xaml/DependencyObjectStore.Binder.cs

```diff
@@ -561,43 +562,69 @@ private void OnDependencyPropertyChanged(DependencyPropertyDetails propertyDetai
 		{
 			SetBindingValue(propertyDetails, args.NewValue);
 
-			var (hasValueInherits, hasValueDoesNotInherit) = GetPropertyInheritanceConfiguration(propertyDetails);
+			GetPropertyInheritanceConfiguration(
+				propertyDetails,
+				args,
+				hasValueInherits: out var hasValueInherits,
+				hasValueDoesNotInherit: out var hasValueDoesNotInherits,
+				valueAsProvider: out var newValueAsProvider
+			);
 
-			if (!hasValueDoesNotInherit && hasValueInherits)
+			if (!hasValueDoesNotInherits)
```

Comment:
This block of conditions starts to get quite hard to follow... I wonder if it'd be more readable with a nested method like:

```csharp
SetChildrenBindable(object? obj) => _childrenBindable[GetOrCreateChildBindablePropertyIndex(propertyDetails.Property)] = obj;
```